### PR TITLE
feat(provider/azure): Support enableIpForwading in networkProfile

### PR DESCRIPTION
**Pull request by EasyAsABC123**
resolves https://github.com/spinnaker/spinnaker/issues/6318

This will break when other changes are pushed to azure-clouddriver since the test will need to support those features as well.

### CODE REVIEW

**Comment by kfrank2017** [comment]
This looks great and like it will provide the needed functionality.

**Review by kfrank2017 — APPROVED** [review]

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy [hunk 1]
```diff
@@ -171,6 +172,15 @@ class AzureServerGroupDescription extends AzureResourceOpsDescription implements
     azureSG.appGatewayName = scaleSet.tags?.appGatewayName
     azureSG.loadBalancerType = azureSG.appGatewayName != null ? AzureLoadBalancer.AzureLoadBalancerType.AZURE_APPLICATION_GATEWAY.toString() : AzureLoadBalancer.AzureLoadBalancerType.AZURE_LOAD_BALANCER.toString()
     azureSG.appGatewayBapId = scaleSet.tags?.appGatewayBapId
+
+    def networkInterfaceConfigurations = scaleSet.virtualMachineProfile()?.networkProfile()?.networkInterfaceConfigurations()
+
+    if (networkInterfaceConfigurations) {
```
Comment (harikumarks) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy:178]: can we use && than nested loops?

**Comment by neil-yechenwei** [comment]
Thanks @EasyAsABC123 for the commit. Could you resolve the conflicts before the PR review? Thanks in advance.

**Comment by EasyAsABC123** [comment]
@neil-yechenwei updated, I must of missed this one.  Feel free to PR any of the others if you run into this in the future.  Thanks!

**Comment by EasyAsABC123** [comment]
```
> Task :clouddriver-kubernetes:integrationTest FAILED


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':clouddriver-kubernetes:integrationTest'.
> There were failing tests. See the report at: file:///home/runner/work/clouddriver/clouddriver/clouddriver-kubernetes/build/reports/tests/integrationTest/index.html
```

Since i didn't touch `clouddriver-kubernetes` i'm going to ignore the failure

**Comment by neil-yechenwei** [comment]
@EasyAsABC123 , after checked the latest [commit](https://github.com/spinnaker/clouddriver/commits/master), seems `Task :clouddriver-kubernetes:integrationTest` works fine. Could you have a retry for the CI?

**Comment by EasyAsABC123** [comment]
@neil-yechenwei rerunning the builds now after another rebase.

**Comment by EasyAsABC123** [comment]
@neil-yechenwei looking good for your approval

**Comment by neil-yechenwei** [comment]
LGTM

**Comment by EasyAsABC123** [comment]
@ajordens Good to merge?

**Pull request merged**